app/flatpak-builtins-alias.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright © 2022 Matthew Leeds
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *       Matthew Leeds <mwleeds@protonmail.com>
+ */
+
+#include "config.h"
+
+#include <locale.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <string.h>
+
+#include <glib/gi18n.h>
+
+#include "libglnx.h"
+
+#include "flatpak-builtins.h"
+#include "flatpak-builtins-utils.h"
+#include "flatpak-cli-transaction.h"
+#include "flatpak-quiet-transaction.h"
+#include "flatpak-utils-private.h"
+#include "flatpak-error.h"
+#include "flatpak-table-printer.h"
+
+static gboolean opt_remove;
+
+static GOptionEntry options[] = {
+  { "remove", 0, 0, G_OPTION_ARG_NONE, &opt_remove, N_("Remove the specified alias"), NULL },
+  { NULL }
+};
+
+gboolean
+flatpak_builtin_alias (int argc, char **argv, GCancellable *cancellable, GError **error)
+{
+  g_autoptr(GOptionContext) context = NULL;
+  g_autoptr(GPtrArray) dirs = NULL;
+  int i;
+
+  context = g_option_context_new (_("ALIAS [APP] - Add an alias for running the app APP"));
+  g_option_context_set_translation_domain (context, GETTEXT_PACKAGE);
+
+  if (!flatpak_option_context_parse (context, options, &argc, &argv,
+                                     FLATPAK_BUILTIN_FLAG_ALL_DIRS | FLATPAK_BUILTIN_FLAG_OPTIONAL_REPO,
+                                     &dirs, cancellable, error))
+    return FALSE;
+
+  /* Move the user dir to the front so it "wins" in case an app is in more than
+   * one installation */
+  if (dirs->len > 1)
+    {
+      /* Walk through the array backwards so we can safely remove */
```

Comment:
This is not really needed as we stop once we find the target item

Reply:
But I see its just copied from flatpak run, so lets just leave it as is for now. Can be cleaned up in both later.